File: android/src/main/java/plugin/kakaologin/knc/kakaoLoginPlugin.java
package plugin.kakaologin.knc;

import android.app.Activity;
import android.util.Log;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.kakao.sdk.common.KakaoSdk;
import com.kakao.sdk.user.UserApiClient;
import com.kakao.sdk.user.model.Account;

@CapacitorPlugin(name = "kakaoLogin")
public class kakaoLoginPlugin extends Plugin {

    private Boolean initSwitch = false;

    @PluginMethod
    public void kakaoLogin(PluginCall call) {
        if (!initSwitch) {
            String appKey = call.getString("appKey");
            try {
                KakaoSdk.init(super.getActivity(), appKey);
            } catch (Exception e) {
                call.reject(e.getMessage());
            }
            initSwitch = true;
        }

        Activity activity = super.getActivity();
        if (UserApiClient.getInstance().isKakaoTalkLoginAvailable(activity)) {
            UserApiClient
                .getInstance()
                .loginWithKakaoTalk(
                    activity,
                    (oAuthToken, error) -> {
                        if (error != null) {
                            call.reject(error.toString());
                        } else if (oAuthToken != null) {
                            UserApiClient
                                .getInstance()
                                .me(
                                    (user, meError) -> {
                                        Account kakaoAccount = user.getKakaoAccount();

                                        String email = kakaoAccount.getEmail();
                                        JSObject ret = new JSObject();
                                        ret.put("email", email);
                                        call.resolve(ret);
                                        return null;
                                    }
                                );
                        } else {
                            call.reject("no_data");
                        }
                        return null;
                    }
                );
        } else {
            UserApiClient
                .getInstance()
                .loginWithKakaoAccount(
                    activity,
                    (oAuthToken, error) -> {
                        if (error != null) {
                            call.reject("error::" + error.toString());
                        } else if (oAuthToken != null) {
                            UserApiClient
                                .getInstance()
                                .me(
                                    (user, meError) -> {
                                        Account kakaoAccount = user.getKakaoAccount();
                                        String email = kakaoAccount.getEmail();
                                        JSObject ret = new JSObject();
                                        ret.put("email", email);
                                        call.resolve(ret);
                                        return null;
                                    }
                                );
                        } else {
                            call.reject("no_data");
                        }
                        return null;
                    }
                );
        }
    }
}

File: ios/Plugin/kakaoLoginPlugin.swift
import Foundation
import Capacitor
import KakaoSDKUser
import KakaoSDKCommon

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(kakaoLoginPlugin)
public class kakaoLoginPlugin: CAPPlugin {
    
    var initSwitch: Bool = false
    

        @objc public func kakaoLogin(_ call: CAPPluginCall) -> Void {
            
            guard let appKey = call.getString("appKey") else {
                call.reject("appKey not provided")
                return
            }
            
            if !initSwitch {
                KakaoSDK.initSDK(appKey:appKey, loggingEnable:true)
                initSwitch = true;
            }
            
    

            
            //카카오톡 설치 여부 확인
            if (UserApi.isKakaoTalkLoginAvailable()) {
                UserApi.shared.loginWithKakaoTalk {(oauthToken, error) in
                    if let error = error {
                        call.reject("error")
                    }
                    else {
                        self.kakaoUserInfo(call)
                    }
                }}
            else{
                // 앱이 설치되어 있지 않아서
                UserApi.shared.loginWithKakaoAccount {(oauthToken, error) in
                    if let error = error {
                        let errorString = String(describing: error)
                        call.reject( errorString)
                    }
                    else {
                        self.kakaoUserInfo(call)
                    }
                }
            }
        }

                public func kakaoUserInfo(_ call: CAPPluginCall) {
            UserApi.shared.me() { (user, error) in
                if let error = error {
                    let errorString = String(describing: error)
                    call.reject( errorString)
                }
                else {
                    call.resolve([
                        "email": user?.kakaoAccount?.email ?? ""
                    ])
                }
            }
        }
}
